### PR TITLE
feat(mrm_handler): add mrm_handler

### PR DESCRIPTION
**Pull request by TetsuKawa**
## Description

This PR adds mrm_handler node that is going to be replaced with emergency_handler node.
The difference from emergency_handler is that mrm_handler node uses operation_mode_availability while emergency_handler uses hazard_status.

## Tests performed

Confirmed that mrm_handler has the same functionality as emergency_handler.
Confirmed that pull_over functionality can be executed.
Confirmed that behavior transitions from pull_over to comfortable_stop and emergency_stop.

## Notes for reviewers

To transition from emergency_handler to mrm_handler,

- Replace the aggregator+system_error_monitor nodes with diagnostic_graph_aggregator node.
- Create a graph configuration file for diagnostic_graph_aggregator.
- Launch new hazard_status_converter node.

## Effects on system behavior

This change not only allows for the execution of various MRM behaviors that could not be expressed in hazard_status, but also allows for extensibility of MRM behaviors.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/mrm_handler/README.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# mrm_handler
+
+## Purpose
+
+MRM Handler is a node to select proper MRM from from system failure state contained in OperationModeAvailability.
```
Comment (veqcc) [inline system/mrm_handler/README.md:5]: MRM Handler is a node to select **a** proper MRM from **a** system failure state

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+bool MrmHandler::isEmergency() const
+{
+  return !operation_mode_availability_->autonomous || is_emergency_holding_;
+}
+
+bool MrmHandler::isArrivedAtGoal()
+{
+  using autoware_adapi_v1_msgs::msg::OperationModeState;
+
+  if (operation_mode_state_->mode == OperationModeState::STOP) {
```
Comment (veqcc) [inline system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:586]: Why not just `return operation_mode_state_->mode == OperationModeState::STOP` ?

File: system/mrm_handler/README.md (hunk 1 above)
Comment (TetsuKawa) [reply system/mrm_handler/README.md:5]: Thanks for your comment！

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp (hunk 2 above)
Comment (TetsuKawa) [reply system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:586]: just as you said！

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+      }
+    }
+  } else {
+    // Emergency
+    // Send recovery events if "not emergency" or "takeover done"
+    if (!is_emergency) {
+      transitionTo(MrmState::NORMAL);
+      return;
+    }
+    // TODO(Kenji Miyake): Check if human can safely override, for example using DSM
```
Comment (veqcc) [inline system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:459]: It looks better to change `TODO(Kenji Miyake)` to someone else

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp [hunk 4]
```diff
@@ -0,0 +1,587 @@
...
+  if (mrm_behavior == MrmState::NONE) {
+    RCLCPP_WARN(this->get_logger(), "MRM behavior is None. Do nothing.");
+    return;
+  }
+  if (mrm_behavior == MrmState::PULL_OVER) {
+    auto result = client_mrm_pull_over_->async_send_request(request).get();
+    if (result->response.success == true) {
+      RCLCPP_WARN(this->get_logger(), "Pull over is operated");
+    } else {
+      RCLCPP_ERROR(this->get_logger(), "Pull over is failed to operate");
```
[the remark this answers is not in the document]
Comment (TetsuKawa) [reply system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp]: Thanks for your recommendation.

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp (hunk 3 above)
Comment (TetsuKawa) [reply system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:459]: I will temporarily change `TODO(Kenji Miyake)` in charge to me.
I would like to discuss later the necessity of the modification and the content of the modification.

File: system/mrm_handler/include/mrm_handler/mrm_handler_core.hpp [hunk 5]
```diff
@@ -0,0 +1,160 @@
+// Copyright 2020 Tier IV, Inc.
```
Comment (veqcc) [inline system/mrm_handler/include/mrm_handler/mrm_handler_core.hpp:1]: Copyright 2024 TIER IV, Inc.

File: system/mrm_handler/src/mrm_handler/mrm_handler_node.cpp [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// Copyright 2024 Tier IV, Inc.
```
Comment (veqcc) [inline system/mrm_handler/src/mrm_handler/mrm_handler_node.cpp:1]: TIER IV

File: system/mrm_handler/README.md [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+| `/vehicle/status/control_mode`         | `autoware_auto_vehicle_msgs::msg::ControlModeReport` | Used to check vehicle mode: autonomous or manual                                                    |
+| `/system/mrm/emergency_stop/status`    | `tier4_system_msgs::msg::MrmBehaviorStatus`          | Used to check if MRM emergency stop operation is available                                          |
+| `/system/mrm/comfortable_stop/status`  | `tier4_system_msgs::msg::MrmBehaviorStatus`          | Used to check if MRM comfortable stop operation is available                                        |
+| `/system/mrm/pull_over_manager/status` | `tier4_system_msgs::msg::MrmBehaviorStatus`          | Used to check if MRM pull over operation is available                                               |
+| `/api/mrm/emergency_stop/status`       | `tier4_system_msgs::msg::MrmBehaviorStatus`          | Used to check if MRM pull over operation is finished                                                |
```
Comment (mkuri) [inline system/mrm_handler/README.md:25]: Is this `/api/operation_mode/state`?

File: system/mrm_handler/README.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+![mrm-state](image/mrm-state.svg)
+
+## Inputs / Outputs
+
+### Input
+
+| Name                                   | Type                                                 | Description                                                                                         |
+| -------------------------------------- | ---------------------------------------------------- | --------------------------------------------------------------------------------------------------- |
+| `/localization/kinematic_state`        | `nav_msgs::msg::Odometry`                            | Used to decide whether vehicle is stopped or not                                                    |
+| `/system/operation_mode/availability`  | `autoware_adapi_v1_msgs::msg::OperationModeState`    | Used to select proper MRM from system available mrm behavior contained in operationModeAvailability |
```
Comment (mkuri) [inline system/mrm_handler/README.md:20]: The type seems to be wrong.

File: system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp [hunk 9]
```diff
@@ -0,0 +1,587 @@
...
+    return;
+  }
+  const bool is_operation_mode_availability_timeout =
+    (this->now() - stamp_operation_mode_availability_).seconds() >
+    param_.timeout_operation_mode_availability;
+  if (is_operation_mode_availability_timeout) {
+    RCLCPP_WARN_THROTTLE(
+      this->get_logger(), *this->get_clock(), std::chrono::milliseconds(1000).count(),
+      "heartbeat operation_mode_availability is timeout");
+    mrm_state_.state = autoware_adapi_v1_msgs::msg::MrmState::MRM_OPERATING;
```
Comment (mkuri) [inline system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:384]: Are you going to apply this change later?
https://github.com/autowarefoundation/autoware.universe/pull/6399

Comment (TetsuKawa) [reply system/mrm_handler/src/mrm_handler/mrm_handler_core.cpp:384]: Yes. This PR just replaces the topic.

File: system/mrm_handler/include/mrm_handler/mrm_handler_core.hpp (hunk 5 above)
Comment (TetsuKawa) [reply system/mrm_handler/include/mrm_handler/mrm_handler_core.hpp]: Thanks for your recommendation！

File: system/mrm_handler/src/mrm_handler/mrm_handler_node.cpp (hunk 6 above)
Comment (TetsuKawa) [reply system/mrm_handler/src/mrm_handler/mrm_handler_node.cpp]: Thanks for your recommendation！

File: system/mrm_handler/README.md (hunk 7 above)
Comment (TetsuKawa) [reply system/mrm_handler/README.md]: Yes. That line is a mistake.
I fixed it. Thanks for your recommendation!

File: system/mrm_handler/README.md (hunk 8 above)
Comment (TetsuKawa) [reply system/mrm_handler/README.md]: I fixed it. Thanks for your recommendation!

File: system/mrm_handler/README.md (hunk 7 above)
Comment (mkuri) [reply system/mrm_handler/README.md]: @TetsuKawa 
The type still seems to be wrong.

Comment (mkuri) [reply system/mrm_handler/README.md]: https://github.com/autowarefoundation/autoware.universe/pull/6400/commits/be3a3da795e3363b6ca3a26a7e47c6bc33a2de96

**Review by mkuri — APPROVED** [review]
LGTM

**Pull request merged**